lib/workers/global/config/parse/home-dir.ts

```diff
@@ -0,0 +1,74 @@
+import os from 'os';
+import ini from 'ini';
+import { join } from 'upath';
+import type { AllConfig } from '../../../../config/types';
+import { logger } from '../../../../logger';
+import { readFile } from '../../../../util/fs';
+
+export interface HomeDirConfig {
+  fileList: string[];
+  config: AllConfig;
+}
+
+export async function readHomeDirFiles(): Promise<HomeDirConfig | null> {
+  const fileList: string[] = [];
+  const config: AllConfig = { hostRules: [], packageRules: [] };
+  const homeDir = os.homedir();
+  const npmrcFileName = join(homeDir, '.npmrc');
+  let npmrcContent: string;
+  try {
+    npmrcContent = await readFile(npmrcFileName, 'utf8');
+  } catch (err) {
+    // Ignore
+  }
+  if (npmrcContent) {
+    fileList.push(npmrcFileName);
+    try {
+      const npmrc = ini.parse(npmrcContent);
+      if (npmrc.registry) {
+        config.packageRules.push({
+          matchDatasources: ['npm'],
+          registryUrls: [npmrc.registry],
+        });
+      }
+      if (npmrc._authToken) {
+        config.hostRules.push({ hostType: 'npm', token: npmrc._authToken });
+      }
+      if (npmrc._auth) {
+        config.hostRules.push({
+          hostType: 'npm',
+          token: npmrc._auth,
+          authType: 'Basic',
+        });
+      }
+      for (const [key, val] of Object.entries(npmrc)) {
+        const [scope, topic] = key.split(':');
+        if (topic === 'registry') {
+          config.packageRules.push({
+            matchDatasources: ['npm'],
+            matchPackagePrefixes: [scope],
+            registryUrls: [val],
```

Comment:
I think this is currently not used by npm datasource?

Reply:
Yes, you're right. Thinking if I could/should change that though. I can't think what held it back before.